core/federated/RTI/rti_lib.h

```diff
@@ -114,26 +75,42 @@ typedef enum clock_sync_stat {
 /**
  * Structure that an RTI instance uses to keep track of its own and its
  * corresponding federates' state.
- */
-typedef struct RTI_instance_t {
-    // RTI's decided stop tag for federates
+ * It is a special case of `enclave_RTI_t` (declared in enclave.h). Inheritence
+ * is mimicked by having the first attributes to be the same as of enclave_RTI_t,
+ * except that enclaves attribute here is of type `federate_t**`, while it
+ * is of type `enclave_t**` in `enclave_RTI_t`.
+ *     // **************** IMPORTANT!!! ********************
+ *     // **   If you make any change to this struct,     **
+ *     // **   you MUST also change  enclave_RTI_t in     **
+ *     // ** (enclave.h)! The change must exactly match.  **
+ *     // **************************************************
+ */
+typedef struct federation_RTI_t {
```

Comment:
Why not use inheritance directly and avoid the problem of having to match the fields exactly?

```
typedef federation_RTI_t {
   enclave_RTI_t base;
   bool tracing_enabled;
   ...
 };
```

Then you can cast federation_RTI_t to enclave_RTI_t or just get it from the `base` argument. There is an awkwardness in having to create `federate_t` objects and store pointers to them in the `enclave_t` pointers in `enclave_RTI_t` and later cast them back to `federate_t`.

Reply:
This is rather a design decision. @edwardalee and I discussed both solutions.
I ended up choosing this one to avoid cast operations as much as possible and get the most of the compiler help. Currently, we have only 1 single cast, compared to a number when implementing inheritance with `enclave_RTI_t base;`. This also helps with segmentation faults and avoids one level of indirection.

Reply:
OK, fair enough